src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs

```diff
@@ -491,7 +491,7 @@ protected override void ApplyDocumentTextChanged(DocumentId documentId, SourceTe
                 {
                     // The edits would get sent by the co-authoring service to the owner.
                     // The invisible editor saves the file on being disposed, which should get reflected  on the owner's side.
-                    using (var invisibleEditor = new InvisibleEditor(_serviceProvider, document.FilePath, hierarchyOpt: null,
+                    using (var invisibleEditor = new InvisibleEditor(_serviceProvider, document.FilePath!, hierarchy: null,
```

Comment:
❔ How do we know `FilePath` is not null here? Is there already validation of this condition earlier in the code path?

Reply:
Ditto: code never could have worked prior. And more practically, remote files are backed by the file system in a certain magic temp directory.